bodhi/tests/client/test_bindings.py

```diff
@@ -987,3 +987,72 @@ def test___unicode__(self):
         self.assertEqual(unicode(exc.update), u'bodhi-2.2.4-1.el7')
         self.assertEqual(type(unicode(exc.update)), unicode)
         self.assertEqual(unicode(exc), 'Update not found: bodhi-2.2.4-1.el7')
+
+
+class TestBodhiClient_parse_file(unittest.TestCase):
+    """
+    Test the BodhiClient.parse_file() method.
+    """
+    @mock.patch('__builtin__.open', create=True)
+    @mock.patch('bodhi.client.bindings.BodhiClient._load_cookies')
+    @mock.patch('bodhi.client.bindings.os.path.exists')
+    def test_parsing_valid_file(self, exists, _load_cookies, mock_open):
+        """
+        Test parsing a valid update template file
+        """
+        s = [
+            "[fedora-workstation-backgrounds-1.1-1.fc26]\n",
+            "# bugfix, security, enhancement, newpackage (required)\n",
+            "type=bugfix\n",
+            "\n",
+            "# testing, stable\n",
+            "request=testing\n",
+            "\n",
+            "# Bug numbers: 1234,9876\n",
+            "bugs=123456,43212\n",
+            "\n",
+            "# Here is where you give an explanation of your update.\n",
+            "notes=Initial Release\n",
+            "\n",
+            "# Enable request automation based on the stable/unstable karma thresholds\n",
+            "autokarma=True\n",
+            "stable_karma=3\n",
+            "unstable_karma=-3\n",
+            "\n",
+            "# Automatically close bugs when this marked as stable\n",
+            "close_bugs=True\n",
+            "\n",
+            "# Suggest that users restart after update\n",
+            "suggest_reboot=False\n",
+            ""]
+
+        exists.return_value = True
+        mock_open.return_value.readline.side_effect = s * 1000
```

Comment:
Ooops - this line was here from some debugging I was doing when I was trying to figure out why Mock wasn't working. Can you replace this line with ```mock_open.return_value.readline.side_effect = s``` before merging? Thanks!